tinygrad/tensor.py

```diff
@@ -987,6 +987,16 @@ def dropout(self, p=0.5) -> Tensor:
   def one_hot(self, num_classes:int) -> Tensor:
     return (self[..., None] == Tensor.arange(num_classes, requires_grad=False, device=self.device)).where(1, 0)
 
+  # nll --> Negative Log Likelihood (linter was throwing in errors...)
+  def nll(self, target:Tensor, weight:Optional[Tensor] = None, reduction:Literal['none', 'mean', 'sum'] = 'mean') -> Tensor:
```

Comment:
Does this match the torch API perfectly? Do you have tests for all the code paths?

Reply:
except for ignore_indexes parameter and the function name, it should match the API perfectly. A mistake on my end that I didn't test if for all code paths (to confirm, do you mean all the different parameters or something else?)

function name should be nll_loss so I'll switch it to that